core/bignum/bit_or_spec.rb

```diff
@@ -24,9 +24,11 @@
   end
 
   it "raises a TypeError when passed a Float" do
-    lambda {
-      bignum_value | bignum_value(0xffff).to_f
-    }.should raise_error(TypeError)
+    not_supported_on :opal do
+      lambda {
+        bignum_value | bignum_value(0xffff).to_f
```

Comment:
Yes the Problem is here that 0xffff is a Numeric and not a Float in Opal. 

Changing bignum_value(0xffff).to_f to something like 9999999.9 would give the possility to test that Opal raises an Error. For Opal this would be the better test.